<?xml version="1.0"?>
<!--
core.launch
Author: Ravi Joshi
Note: Do not run this file directly. Please call run.launch file instead.
Date: 2020/03/02
-->
<launch>
  <!-- note that the following parameters are going to assigned by other launch file upon call. -->
  <arg name="print" default="" />
  <arg name="frame_id" default="" />
  <arg name="no_depth" default="" />
  <arg name="skeleton" default="" />
  <arg name="pub_topic" default="" />
  <arg name="color_topic" default="" />
  <arg name="depth_topic" default="" />
  <arg name="id_text_size" default="" />
  <arg name="openpose_args" default="" />
  <arg name="cam_info_topic" default="" />
  <arg name="id_text_offset" default="" />
  <arg name="skeleton_hands" default="" />
  <arg name="skeleton_line_width" default="" />
  <arg name="synchronous" default="" />
  <arg name="py_openpose_path" default="" />

  <group if="$(arg synchronous)" >
    <!-- invoke synchronous node -->
    <node name="rosOpenpose" pkg="ros_openpose" type="ros_openpose_synchronous_hand.py" output="screen" required="true" args="$(arg openpose_args)" >
      <param name="frame_id" value="$(arg frame_id)" />
      <param name="no_depth" value="$(arg no_depth)" />
      <param name="pub_topic" value="$(arg pub_topic)" />
      <param name="color_topic" value="$(arg color_topic)" />
      <param name="depth_topic" value="$(arg depth_topic)" />
      <param name="cam_info_topic" value="$(arg cam_info_topic)" />
      <param name="py_openpose_path" value="$(arg py_openpose_path)" />
      <param name="skeleton_line_width" value="$(arg skeleton_line_width)" />
    </node>
  </group>

 <!-- print ros message to terminal -->
  <group if="$(arg print)" >
    <node name="echo" pkg="ros_openpose" type="echo.py" output="screen" required="true" >
      <param name="pub_topic" value="$(arg pub_topic)" />
    </node>
  </group>
</launch>
